src/Tasks/Microsoft.NET.Build.Tasks/GenerateDepsFile.cs

```diff
@@ -163,12 +176,137 @@ protected override void ExecuteCore()
+
+            Dictionary<string, SingleProjectInfo> referenceProjects = SingleProjectInfo.CreateProjectReferenceInfos(
+                ReferencePaths,
+                ReferenceDependencyPaths,
+                ReferenceSatellitePaths);
+
+            IEnumerable<string> excludeFromPublishAssets = PackageReferenceConverter.GetPackageIds(ExcludeFromPublishPackageReferences);
+
+            IEnumerable<RuntimePackAssetInfo> runtimePackAssets =
+                RuntimePackAssets.Select(item => RuntimePackAssetInfo.FromItem(item));
 
+
+            ProjectContext projectContext = lockFile.CreateProjectContext(
+                NuGetUtils.ParseFrameworkName(TargetFramework),
+                RuntimeIdentifier,
+                PlatformLibraryName,
+                RuntimeFrameworks,
+                IsSelfContained);
+
+            var builder = new DependencyContextBuilder2(mainProject, projectContext, IncludeRuntimeFileVersions);
+
+            builder = builder
+                .WithMainProjectInDepsFile(IncludeMainProject)
+                .WithReferenceAssemblies(referenceAssemblyInfos)
+                .WithDirectReferences(directReferences)
+                .WithDependencyReferences(dependencyReferences)
+                .WithReferenceProjectInfos(referenceProjects)
+                .WithExcludeFromPublishAssets(excludeFromPublishAssets)
+                .WithRuntimePackAssets(runtimePackAssets)
+                .WithCompilationOptions(compilationOptions)
+                .WithReferenceAssembliesPath(FrameworkReferenceResolver.GetDefaultReferenceAssembliesPath())
+                .WithPackagesThatWereFiltered(GetFilteredPackages());
+
+            if (CompileReferences.Length > 0)
+            {
+                builder = builder.WithCompileReferences(ReferenceInfo.CreateReferenceInfos(CompileReferences));
+            }
+
+            var resolvedNuGetFiles = ResolvedNuGetFiles.Select(f => new ResolvedFile(f, false))
+                                .Concat(ResolvedRuntimeTargetsFiles.Select(f => new ResolvedFile(f, true)));
+            builder = builder.WithResolvedNuGetFiles(resolvedNuGetFiles);
+
+            DependencyContext dependencyContext = builder.Build();
+
+            var writer = new DependencyContextWriter();
+            using (var fileStream = File.Create(depsFilePath))
+            {
+                writer.Write(dependencyContext, fileStream);
+            }
+            _filesWritten.Add(new TaskItem(depsFilePath));
+        }
+
+        bool _loggedLocalError = false;
+
+        public override bool Execute()
+        {
+            if (!base.Execute() || _loggedLocalError)
+            {
+                return false;
+            }
+            return true;
+        }
+        protected override void ExecuteCore()
+        {
+            if (DepsFileGenerationMode.Equals("old", StringComparison.InvariantCultureIgnoreCase))
```

Comment:
What do you think about:

* V1
* V2
* All

instead of 

* old
* new
* both

?

If we ever wanted to add more "modes", the current strings seem limiting.

Reply:
That's not a bad suggestion, but we don't expect for these to live for long.  The only reason we are including multiple modes is in order to help validate that the new logic behaves the same as the old, and to offer an escape hatch if there is a bug in the new logic.  After a few previews, we plan to remove the old logic altogether.

Reply:
I've filed #3098 to track removing the old logic.